fv3core/stencils/d_sw.py

```diff
@@ -699,62 +656,42 @@ def d_sw(
         spec.namelist, spec.namelist.hord_tm, cache_key="d_sw-tm"
     )
     ra_x, ra_y = fxadv.compute(uc, vc, ut, vt, xfx, yfx, crx, cry, dt)
-    for kstart, nk in k_bounds():
-        fvtp2d_dp(
-            delp,
-            crx,
-            cry,
-            xfx,
-            yfx,
-            ra_x,
-            ra_y,
-            fx,
-            fy,
-            kstart=kstart,
-            nk=nk,
-            nord=column_namelist[kstart]["nord_v"],
-            damp_c=column_namelist[kstart]["damp_vt"],
-        )
 
-    fluxcap.compute(cx, cy, xflux, yflux, crx, cry, fx, fy)
-    initialize_heat_source(heat_s, diss_e)
+    fvtp2d_dp(
+        delp,
+        crx,
+        cry,
+        xfx,
+        yfx,
+        ra_x,
+        ra_y,
+        fx,
+        fy,
+        nord=column_namelist["nord_v"],
+        damp_c=column_namelist["damp_vt"],
+    )
+
+    flux_capacitor(
+        cx,
+        cy,
+        xflux,
+        yflux,
+        crx,
+        cry,
+        fx,
+        fy,
+        origin=spec.grid.full_origin(),
+        domain=spec.grid.domain_shape_full(),
+    )
```

Comment:
Can/should these two be merged (in this PR or later)? `flux_capacitor` is a very simple stencil. Is this blocked by needing to be able to validate subsets?

Reply:
these two can't be merged without making the stencil more confusing. fvtp2d is an object and gets called a lot and flux_capacitor does not always follow it, as well. a future PR could try to do it with a conditional or something